opencensus/stats/exporters/stackdriver_exporter.py

```diff
@@ -0,0 +1,415 @@
+# Copyright 2017, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+import os
+import time
+import platform
+from . import base
+from google.cloud import monitoring_v3
+from opencensus.stats import view_data
+from opencensus.stats import aggregation
+from opencensus.stats import aggregation_data
+from opencensus.stats import stats
+from opencensus.stats import measure
+
+MAX_TIME_SERIES_PER_UPLOAD = 200
+OPENCENSUS_TASK_DESCRIPTION = "Opencensus task identifier"
+DEFAULT_DISPLAY_NAME_PREFIX = "OpenCensus"
+ERROR_BLANK_PROJECT_ID = "expecting a non-blank ProjectID"
+CONS_NAME = "name"
+CONS_TIME_SERIES = "timeseries"
+
+class Options(object):
+	""" Options contains options for configuring the exporter. """
+	def __init__(self, 
+				project_id = "", 
+				on_error = None, 
+				monitoring_client_opt = None, 
+				trace_client_opt = None, 
+				bundle_delay_thrs = None, 
+				bundle_count_thrs = 0, 
+				resource = None, 
+				metric_prefix = "", 
+				default_trace_attributes = [], 
+				default_monitoring_labels = None ):
+		self._project_id = project_id
+		self._on_error = on_error
+		self._monitoring_client_options = monitoring_client_opt
+		self._trace_client_options = trace_client_opt
+		self._bundle_delay_threshold = bundle_delay_thrs
+		self._bundle_count_threshold = bundle_count_thrs
+		self._resource = resource
+		self._metric_prefix = metric_prefix
+		self._default_trace_attributes = default_trace_attributes
+		self._default_monitoring_labels = default_monitoring_labels
+
+	@property
+	def project_id(self):
```

Comment:
just note that there's no need for `@property` decorator if you don't add any extra logic when setting/getting it.
it isn't really pythonic to do so in this situation.
also you should just name the attributes without the underscore.
for more info:
https://docs.quantifiedcode.com/python-anti-patterns/correctness/implementing_java-style_getters_and_setters.html?highlight=java

Reply:
Hi @zeako , thanks for your suggestions.

About the attributes names, what do you suggest ?

I created them with underscore because this is the way that some attributes were created inside view_data, for example.

Reply:
just use their names as-is, no underscore needed or `@property` declarations.

if you also think `namedtuple` can be used instead of just a class, then go for it (though keep in mind that it'll make `Options` immutable, which isn't a bad thing by itself).
and if you want to add functions to it you can always extend it with a class